src/main/java/org/jitsi/jigasi/transcription/OracleTranscriptionService.java

```diff
@@ -0,0 +1,353 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.jigasi.transcription;
+
+import com.fasterxml.uuid.*;
+import com.oracle.bmc.*;
+import com.oracle.bmc.aispeech.model.*;
+import com.oracle.bmc.auth.*;
+import org.jitsi.impl.neomedia.device.*;
+import org.jitsi.jigasi.*;
+import org.jitsi.jigasi.transcription.oracle.*;
+import org.jitsi.utils.logging.*;
+
+import java.io.*;
+import java.time.*;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.function.*;
+
+
+/**
+ * Implements a TranscriptionService which uses the Oracle Cloud Infrastructure
+ * to perform live transcriptions.
+ *
+ * @author Purdel Razvan
+ */
+
+public class OracleTranscriptionService
+        extends AbstractTranscriptionService
+{
+
+    /**
+     * Uses the PCMAudioSilenceMediaDevice which transcodes to 16khz PCM audio
+     *
+     * @param listener
+     * @return AudioMixerMediaDevice
+     */
+    @Override
+    public AudioMixerMediaDevice getMediaDevice(ReceiveStreamBufferListener listener)
+    {
+        if (this.mediaDevice == null)
+        {
+            this.mediaDevice = new TranscribingAudioMixerMediaDevice(new PCMAudioSilenceMediaDevice(), listener);
+        }
+
+        return this.mediaDevice;
+    }
+
+    /**
+     * The logger for this class
+     */
+    private final static Logger logger
+            = Logger.getLogger(OracleTranscriptionService.class);
+
+    /**
+     * The config key of the websocket to the speech-to-text service.
+     */
+    public final static String WEBSOCKET_URL
+            = "org.jitsi.jigasi.transcription.oci.websocketUrl";
+
+    public final static String OCI_AUTH_CONFIGURATION_FILE
+            = "org.jitsi.jigasi.transcription.oci.configFile";
+
+    public final static String defaultOCIAuthConfFile = "~/.oci/config";
+
+    private static final String configFilePath = JigasiBundleActivator.getConfigurationService()
+            .getString(OCI_AUTH_CONFIGURATION_FILE, defaultOCIAuthConfFile);
+
+    public final static String COMPARTMENT_ID
+            = "org.jitsi.jigasi.transcription.oci.compartmentId";
+
+    private final String compartmentId;
+
+    public final static String DEFAULT_WEBSOCKET_URL = "ws://localhost:8000/ws";
+
+    private static BasicAuthenticationDetailsProvider authProvider;
+
+    static
+    {
+
+        try
+        {
+            authProvider = new ConfigFileAuthenticationDetailsProvider(ConfigFileReader.parse(configFilePath));
```

Comment:
This runs unconditionally if the class is loaded. Can we have it load only if a flag is enabled? I don't see OracleTranscriptionService being instantiated anywhere, so not sure how that's supposed to work.

Reply:
moved it in the constructor